drivers/overlay/overlay.go

```diff
@@ -1167,6 +1159,18 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 		mountTarget = path.Join(id, "merged")
 	}
 	flags, data := mount.ParseOptions(mountData)
+
+	// overlay has a check in place to prevent mounting the same file system twice
+	// if volatile was already specified.
+	err = os.RemoveAll(filepath.Join(workdir, "work/incompat/volatile"))
```

Comment:
Is this correct?  It is going to remove $ID/work/work/incompat/volatile?

Reply:
yes, I got confused in the first version as it removed "$ID/work/incompat/volatile"

Reply:
Should we just fix what we created rather then what we remove?

Reply:
this file is created by the overlay driver in the kernel, we have no control over it: https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html?highlight=overlayfs#volatile-mount:

```
When overlay is mounted with “volatile” option, the directory “$workdir/work/incompat/volatile” is created. During next mount, overlay checks for this directory and refuses to mount if present. This is a strong indicator that user should throw away upper and work directories and create fresh one. In very limited cases where the user knows that the system has not crashed and contents of upperdir are intact, The “volatile” directory can be removed.
```